File: cmdtest.py
from typing import *
import sys
import json
from CmdParser import CmdParser

if __name__=="__main__":
	cmd:CmdParser
	try:
		cmd=CmdParser(["help","version"],["alpha","beta","gamma","alpaca"])
	except Exception as ex:
		print(f'{{"python":{{"{type(ex).__name__}":"{str(ex)}"}}}}')
		sys.exit()
	if cmd.isDefault:
		print('{"python":"デフォルトオプションで実行しています。"}')
		sys.exit()

	print('{"python":{"args":%s,"flgs":%s,"opts":%s}}'%(
		json.dumps(cmd.args,ensure_ascii=False,separators=(",",":")),
		json.dumps(cmd.flgs,ensure_ascii=False,separators=(",",":")),
		json.dumps(cmd.opts,ensure_ascii=False,separators=(",",":"))))

File: cmdtest.ts
"use strict";
import {CmdParser} from "./CmdParser";

(function(){
	var cmd:CmdParser;
	try{
		cmd=new CmdParser(["help","version"],["alpha","beta","gamma","alpaca"]);
	}
	catch(ex){
		console.log(`{"nodejs":{"${ex.name}":"${ex.message}"}}`);
		return;
	}
	if(cmd.isDefault){
		console.log('{"nodejs":"デフォルトオプションで実行しています。"}');
		return;
	}
	console.log('{"nodejs":{"args":%s,"flgs":%s,"opts":%s}}',
		JSON.stringify(cmd.args),
		JSON.stringify(Object.fromEntries([...cmd.flgs])),
		JSON.stringify(Object.fromEntries([...cmd.opts])));
})();
